homeassistant/components/sensor/__init__.py

```diff
@@ -266,9 +266,7 @@ def numeric_state_expected(self) -> bool:
         with suppress(ValueError):
             # Custom device classes are not considered numeric
             device_class = SensorDeviceClass(str(self.device_class))
-        if device_class and device_class not in _NON_NUMERIC_DEVICE_CLASSES:
-            return True
-        return False
+        return device_class is None and device_class not in _NON_NUMERIC_DEVICE_CLASSES
```

Comment:
Reverted this change as it is not the same logic.

Reply:
It should be `device_class is not None`

Reply:
That, sorry, it was a quick untested suggestion on GitHub

Reply:
> `return device_class is None or device_class not in NON_NUMERIC_DEVICE_CLASSES`
Fails too in tests :-(. the bool wrapper ensures we return a bool